### ISSUE TITLE
Creating JSON report fails with exit code 2

### ISSUE THREAD

**Issue by p3k** [stylelint/stylelint#4425]
> Clearly describe the bug

I added a script to `package.json` for [creating a report in JSON format](https://community.sonarsource.com/t/external-analyzer-reports-report-generation-for-various-languages/9791) in order to [pipe it into Sonarqube as external issues](https://docs.sonarqube.org/latest/analysis/external-issues/):

```json
{
  "scripts": {
    "stylelint": "stylelint -q -f json **/*.less > stylelint-report.json"
  }
}
```

When running the script with `yarn run stylelint` I get the following output:

```plain
→ yarn run stylelint
yarn run v1.19.1
$ stylelint -q -f json **/*.less > stylelint-report.json
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
``` 

The problem starts with the line `error Command failed with exit code 2`. 

At this point our build framwork (Gradle) is going to throw an error on its behalf and the whole analysis is stopped. Thus, it never comes to the inclusion of the generated report in Sonarqube.

> Which rule, if any, is the bug related to?

None according to my knowledge.

> What CSS is needed to reproduce the bug?

Any CSS having an issue with severity “error”.

> What stylelint configuration is needed to reproduce the bug?

I am not sure the configuration is influencing the behavior. Nevertheless this is the one I am using:

```js
module.exports = {
        'extends': 'stylelint-config-standard',
        'rules': {
                'indentation': 'tab',
                'selector-class-pattern': '^([a-z][a-z0-9]*)(-[a-z0-9]+)*$'
        }
};
```

> Which version of stylelint are you using?

`12.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Gradle YarnTask running the script in package.json as detailed above.

```groovy
task runStyleLint(type: YarnTask) {
	inputs.file 'package.json'
	outputs.files "${project.buildDir}/stylelint-report.json"
	args = ['run', 'stylelint']
}
```

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

I don’t think so.

> What did you expect to happen?

Exit code 0.

> What actually happened (e.g. what warnings or errors did you get)?

Exit code 2.

I currently work around this behavior by adding the following code to the Gradle task:

```groovy
execOverrides {
	it.ignoreExitValue = true
}
``` 

However, I think StyleLint should either
* simply return 0 as exit code when creating a report 
* or there should be a command-line switch ignoring the linting result on exit.

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

Please, create repository with reproducible problem.

**Comment by nikolay-borzov** [comment]
I have the same problem when running `stylint` with `--output-file` specified.
`stylelint` outputs the result to the console and returns 2 exit code.

I think it better to output nothing if `--output-file` is specified and always return 0 as exit code. At least ESLint CLI behaves in that way.

**Comment by jeddy3** [comment]
Closing for now. Feel free to reopen with a repository with reproducible problem.
